### PR TITLE
Cloud-to-ground lightning flash frequency coupling

### PR DESCRIPTION
**Pull request by fvitt**
### Description of changes

Enables passing prognostic cloud-to-ground lightning flash frequency from CAM to CTSM

### Specific notes

Issues Fixed (include github issue #):
 https://github.com/ESCOMP/CTSM/issues/1630 
 https://github.com/ESCOMP/CAM/issues/567

Are changes expected to change answers? (specify if bfb, different at roundoff, more substantial) 
b4b

### Testing performed

  ERC_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.cheyenne_intel.cam-outfrq3s_cosp 
  SMS_D_Ln9_Vnuopc.f19_f19_mg17.FWma2000climo.cheyenne_intel.cam-outfrq9s
  SMS_D_Ln9_Vnuopc_P720x1.ne30pg3_ne30pg3_mg17.FCLTHIST.cheyenne_intel.cam-outfrq9s 
  SMS_Ln9_Vnuopc.f19_f19.F2000climo.cheyenne_intel.cam-silhs

### CODE REVIEW

File: cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90 [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    rc = ESMF_SUCCESS
+
+    !--- Open and read namelist ---
+    if ( len_trim(NLFilename) == 0 ) then
+       call shr_sys_abort( subname//'ERROR: nlfilename not set' )
+    end if
+    call shr_log_getLogUnit(s_logunit)
+    call ESMF_VMGetCurrent(vm, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
```
Comment (jedwards4b) [inline cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90:62]: Please change these to use the helper routine
 if (chkerr(rc,__LINE__,u_FILE_u)) return

Comment (jedwards4b) [reply cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90:62]: It's in:  use nuopc_shr_methods, only : chkerr

File: cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90 [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  !========================================================================
+  ! Module for handling namelist variables related to lightning coupling
+  !========================================================================
+
+  use ESMF         , only : ESMF_VMGetCurrent, ESMF_VM, ESMF_VMGet
+  use ESMF         , only : ESMF_LogFoundError, ESMF_LOGERR_PASSTHRU, ESMF_SUCCESS
+  use shr_sys_mod  , only : shr_sys_abort
+  use shr_log_mod  , only : shr_log_getLogUnit
+  use shr_nl_mod   , only : shr_nl_find_group_name
+  use shr_mpi_mod  , only : shr_mpi_bcast
```
Comment (jedwards4b) [inline cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90:12]: We are deprecating shr_mpi_mod, please use ESMF_VMBroadcast instead.

**Review by jedwards4b — CHANGES_REQUESTED** [review]

File: cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90 (hunk 1 above)
Comment (fvitt) [reply cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90]: fixed

File: cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90 [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+       end if
+
+       atm_provides_lightning_out = atm_provides_lightning
+
+    end if
+
+    ! ------------------------------------------------------------------------
+    ! Broadcast values to all processors
+    ! ------------------------------------------------------------------------
+    call shr_mpi_bcast(atm_provides_lightning_out, mpicom)
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90]: fixed

File: cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90 (hunk 2 above)
Comment (fvitt) [reply cesm/nuopc_cap_share/shr_lightning_coupling_mod.F90]: fixed

**Review by jedwards4b — APPROVED** [review]
one more minor change requested and this should be good to go.

**Pull request merged**